runtime/src/main/java/org/corfudb/protocols/wireprotocol/InboundMsgFilterHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.corfudb.protocols.wireprotocol;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represent a Netty handler in charge of applying filters on the inbound
+ * Corfu messages in order to transform them. The transformation are supplied
+ * to this handler as a {@link List} of {@link MsgHandlingFilter}s which will
+ * be applied in order. An example of such transforming filters is drop certain
+ * messages.
+ *
+ * Created by Sam Behnam on 5/18/18.
+ */
+@Slf4j
+public class InboundMsgFilterHandler extends ChannelDuplexHandler {
+
+    private final List<MsgHandlingFilter> handlingFilters;
+
+    public InboundMsgFilterHandler() {
```

Comment:
The handlingFilters is final. This constructor initializes it to an empty list preventing us from adding any rules. Is this constructor required?

Reply:
Done